### ISSUE TITLE
[MacOS] New tab resets window to maximized

### ISSUE THREAD

**Issue by TitanNano** [raphamorim/rio#686]
# Steps to reproduce
1. Set` window.mode` to `maximized`
2. Resize Window
3. Open new tab

# Expected outcome
Window size and position remains as is.

# Actual outcome
Window is resized back to maximize

---

Platform: macOS 15.0
Version: 0.15.0

**Comment by raphamorim** [comment]
Hey @TitanNano thank you for the issue, is this issue something introduced on 0.1.15 or first time that you got this?

**Comment by TitanNano** [comment]
@raphamorim I'm not sure. I just started using the new tiling feature of macOS 15.0 and noticed this issue.

**Comment by johnend** [comment]
Yep, I have experience something related using Aerospace for tiling. Rather than setting the window to maximised, if the tabbing is set to "NativeTab" Aerospace acts as if there are two windows, rather than one with tabs.

Creating a new "Bookmark" tab for me opens a new window my assumption is that is also not playing well with tiling.

**Comment by TitanNano** [comment]
@raphamorim I was able to reproduce this issue as far back as `0.0.25`. (anything earlier than that wouldn't run properly anymore)
